### PR TITLE
Add kernelCTF CVE-2023-31436

### PR DESCRIPTION
**Pull request by liona24**

### CODE REVIEW

File: pocs/linux/kernelctf/CVE-2023-31436_mitigation/metadata.json [hunk 1]
```diff
@@ -0,0 +1,24 @@
+{
+    "$schema": "https://google.github.io/security-research/kernelctf/metadata.schema.v1.json",
+    "submission_id": "exp52",
```
Comment (koczkatamas) [inline pocs/linux/kernelctf/CVE-2023-31436_mitigation/metadata.json:3]: In v2, this is an array:
```suggestion
    "submission_ids": ["exp52"],
```

**Review by koczkatamas — CHANGES_REQUESTED** [review]
We switched JSON schema version, please accept these changes to make the PR check succeed.

**Comment by liona24** [comment]
Hey. Thanks for the update. My Makefile was using a slightly different target, so I fixed that. The exploit is using side channels for KASLR bypass. This does not appear to be working on the runner, does however work on the real instances. (tbh I haven't checked in a while, where there any infra changes?). Any suggestions on how to proceed here? I can have a closer look in a week or so.

**Comment by liona24** [comment]
So I checked again, the cache timing side channel works fine on the real instance. Both use QEMUs `-cpu host`, not sure what is underneath here.

Would KASLR disabled be an option for the CI job? As far as the CTF is concerned KASLR bruteforce is a valid strategy anyway, so you probably do not want such submissions to spam your runner too.

Also the CLA check is failing because apparently you (@koczkatamas  are not approved as a contributor and we shared some commits.

**Comment by koczkatamas** [comment]
Hey!

Yeah, we will provide an option to disable KASLR. When it's ready (ETA:
next week), I will comment here.

**Comment by koczkatamas** [comment]
Implemented, KASLR base will be supplied as the first argument to `./exploit`. Example full command line: `su user -c '/tmp/exp/exploit ffffffff8a000000'`.

You need to change to the v3 metadata schema and modify the exploits node to this:

```json
    "exploits": {
        "mitigation-6.1": {
            "uses": ["userns"],
            "stability_notes": "5% success rate",
            "requires_separate_kaslr_leak": true
        }
    }
```

See the full modified file here: https://github.com/koczkatamas2/security-research/pull/1/files#diff-a3e5b8f8744c0ae7ff2ed99f0c28cb0a940bf5b6964a86eeecc0fa42d19bdd6d

**Comment by liona24** [comment]
Alright thank you very much. The only thing missing now is the CLA check because of the co-authored commits.

**Pull request merged**